src/lib/pubkey/sphincsplus/sphincsplus_common/sp_hash.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPHINCS+ Hashes
+ * (C) 2023 Jack Lloyd
+ *     2023 Fabian Albert, René Meusel, Amos Treiber - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ **/
+
+#include <botan/internal/sp_hash.h>
+
+#include <botan/internal/stl_util.h>
+
+#include <botan/assert.h>
+#include <botan/exceptn.h>
+#include <botan/hash.h>
+#include <botan/sp_parameters.h>
+
+#if defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE)
+   #include <botan/internal/sp_hash_shake.h>
+#endif
+
+#if defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2)
+   #include <botan/internal/sp_hash_sha2.h>
+#endif
+
+#include <memory>
+
+namespace Botan {
+
+Sphincs_Hash_Functions::Sphincs_Hash_Functions(const Sphincs_Parameters& sphincs_params,
+                                               const SphincsPublicSeed& pub_seed) :
+      m_sphincs_params(sphincs_params), m_pub_seed(pub_seed) {}
+
+std::unique_ptr<Sphincs_Hash_Functions> Sphincs_Hash_Functions::create(
+   const Sphincs_Parameters& sphincs_params, [[maybe_unused]] const SphincsPublicSeed& pub_seed) {
```

Comment:
OK another ``[[maybe_unused]]`, and here I do think it must be used since it is used if either SHA2 or SHAKE is enabled.

Reply:
Agreed, thanks! I fixed it.